src/obj.rs

```diff
@@ -139,15 +266,68 @@ where
     }
 }
 
-pub struct Obj<'a, P>
-where
-    P: 'a + GenPolygon,
-{
+impl<P: WriteToBuf<Error = ObjError>> WriteToBuf for Group<P> {
+    type Error = ObjError;
+    /// Serialize this `Group` into the given writer.
+    fn write_to_buf<W: Write>(&self, out: &mut W) -> Result<(), ObjError> {
+        // When index is greater than 0, we know that this group is the same as the previous group,
+        // so don't bother declaring a new one.
+        if self.index == 0 {
+            writeln!(out, "g {}", self.name)?;
+        }
+
+        match &self.material {
+            Some(ObjMaterial::Ref(name)) => writeln!(out, "usemtl {}", name)?,
+            Some(ObjMaterial::Mtl(mtl)) => writeln!(out, "usemtl {}", mtl.name)?,
+            None => {}
+        }
+
+        for poly in &self.polys {
+            poly.write_to_buf(out)?;
+        }
+
+        Ok(())
+    }
+}
+
+/// The data model associated with each `Obj` file.
+#[derive(Debug, PartialEq)]
+pub struct ObjData<P = SimplePolygon> {
```

Comment:
why do we need the default type to be set?

Reply:
Solely trying to make the calls to `load` and `save` less verbose. To someone just trying to load an `Obj` it seems like unnecessary complexity to have to think about what type of polygons to use. That's just my opinion though, I am not sure about the tradeoffs here.

The default parameter here makes this call:
```rust
let obj = Obj::<SimplePolygon>::load(&Path::new("file.obj"))?;
```
into this:
```rust
let obj: Obj = Obj::load(&Path::new("file.obj"))?;
```

Reply:
I'm totally fine with the generic for `load` and `save` methods specifically. Does it have to be put on the whole struct? If yes, we could have an `fn load_simple` and stuff.